### PR TITLE
Add session scoped fixture variants for `capsys`, `conda_cli`, and `tmp_env`

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Occasionally we desire to generate conda environments that can be shared between several tests (e.g., https://github.com/conda/conda-libmamba-solver/pull/524). Previously this was not possible with `tmp_env` since it was a function scoped fixture. In order to support a session scoped fixture variant of `tmp_env` we also needed session scoped fixture variants for `conda_cli` and `capsys`.

We also opt to move all of the fixtures defined in `conda.testing` to the more appropriate `conda.testing.fixtures` module.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda/blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Review by beeankha — DISMISSED** [review]

**Comment by kenodegard** [comment]
@beeankha oops! I forgot to remove the `conda.testing` plugin from `conftest.py`!

**Comment by beeankha** [comment]
> @beeankha oops! I forgot to remove the conda.testing plugin from conftest.py!

Ah good catch!

**Review by beeankha — DISMISSED** [review]
Thank you for implementing these testing-related improvements!

**Comment by beeankha** [comment]
Should the docs examples be updated to reflect where the fixtures are imported from after this change? (e.g. from [`/conda/conda/tree/main/docs/source/dev-guide/writing-tests/integration-tests.md`](https://github.com/conda/conda/blob/main/docs/source/dev-guide/writing-tests/integration-tests.md) [line 23](https://github.com/conda/conda/blob/main/docs/source/dev-guide/writing-tests/integration-tests.md?plain=1#L23), [line 87](https://github.com/conda/conda/blob/main/docs/source/dev-guide/writing-tests/integration-tests.md?plain=1#L87), [line 111](https://github.com/conda/conda/blob/main/docs/source/dev-guide/writing-tests/integration-tests.md?plain=1#L111), [line 156](https://github.com/conda/conda/blob/main/docs/source/dev-guide/writing-tests/integration-tests.md?plain=1#L156) and possibly more)

**Comment by kenodegard** [comment]
@beeankha another great catch! corrected all of the type imports in the test suite too!

**Review by beeankha — DISMISSED** [review]

File: conda/testing/fixtures.py [hunk 1]
```diff
@@ -161,3 +186,287 @@ def _solver_helper(
     assert context.solver == solver
 
     yield solver
+
+
+@pytest.fixture(scope="session")
+def global_capsys(request) -> Iterator[MultiCapture]:
```
Comment (jezdez) [inline conda/testing/fixtures.py:192]:
```suggestion
def session_capsys(request) -> Iterator[MultiCapture]:
```

"global" having a particular meaning in Python might not be what we want in this case

**Review by jezdez — DISMISSED** [review]
Besides the naming, this seems reasonable

**Review by beeankha — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Pull request merged**